backend/src/main/java/fk/prof/backend/http/BackendHttpVerticle.java

```diff
@@ -0,0 +1,177 @@
+import fk.prof.backend.http.handler.RecordedProfileRequestHandler;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.core.shareddata.LocalMap;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.ext.web.handler.LoggerHandler;
+import recording.Recorder;
+
+public class BackendHttpVerticle extends AbstractVerticle {
+  private static Logger logger = LoggerFactory.getLogger(BackendHttpVerticle.class);
+
+  private final JsonObject backendHttpServerConfig;
+  private final JsonObject httpClientConfig;
+  private final LeaderDiscoveryStore leaderDiscoveryStore;
+  private final IProfileWorkService profileWorkService;
+  private final int leaderPort;
+  private Long loadReportTimer = null;
+
+  private LocalMap<Long, Boolean> workIdsInPipeline;
+  private ConfigurableHttpClient httpClient;
+
+  public BackendHttpVerticle(JsonObject backendHttpServerConfig,
+                             JsonObject httpClientConfig,
+                             int leaderPort,
+                             LeaderDiscoveryStore leaderDiscoveryStore,
+                             IProfileWorkService profileWorkService) {
+    this.backendHttpServerConfig = backendHttpServerConfig;
+    this.httpClientConfig = httpClientConfig;
+    this.leaderDiscoveryStore = leaderDiscoveryStore;
+    this.profileWorkService = profileWorkService;
+    this.leaderPort = leaderPort;
+  }
+
+  @Override
+  public void start(Future<Void> fut) {
+    httpClient = ConfigurableHttpClient.newBuilder()
+        .keepAlive(httpClientConfig.getBoolean("keepalive", true))
+        .useCompression(httpClientConfig.getBoolean("compression", true))
+        .setConnectTimeoutInMs(httpClientConfig.getInteger("connect.timeout.ms", 5000))
+        .setIdleTimeoutInSeconds(httpClientConfig.getInteger("idle.timeout.secs", 10))
+        .setMaxAttempts(httpClientConfig.getInteger("max.attempts", 3))
+        .build(vertx);
+
+    Router router = setupRouting();
+    workIdsInPipeline = vertx.sharedData().getLocalMap("WORK_ID_PIPELINE");
+    vertx.createHttpServer(HttpHelper.getHttpServerOptions(backendHttpServerConfig))
+        .requestHandler(router::accept)
+        .listen(backendHttpServerConfig.getInteger("port"), http -> completeStartup(http, fut));
+  }
+
+  private Router setupRouting() {
+    Router router = Router.router(vertx);
+    router.route().handler(LoggerHandler.create());
+
+    router.post(ApiPathConstants.AGGREGATOR_POST_PROFILE).handler(this::handlePostProfile);
+
+    router.post(ApiPathConstants.BACKEND_PUT_ASSOCIATION)
```

Comment:
Not really, it returns a new router impl. Look at io.vertx.ext.web.impl.RouterImpl.java. Also refer to docs here: http://vertx.io/docs/vertx-web/java/#_route_order